client/reader/sidebar/reader-sidebar-tags/index.jsx

```diff
@@ -43,6 +44,15 @@ export class ReaderSidebarTags extends Component {
 		this.setState( ( state ) => ( { addTagCounter: state.addTagCounter + 1 } ) );
 	};
 
+	selectMenu = () => {
+		const { onClick, tags, isOpen } = this.props;
+		if ( ! isOpen ) {
+			onClick();
+		}
+		const defaultSelection = tags?.length ? `/tag/${ tags[ 0 ]?.slug }` : '/tags';
+		page.redirect( defaultSelection );
```

Comment:
I will update this to use `page` instead of `page.redirect` and only call it if we aren't on that route. This should help with browser history working more as expected.